library/Freckle/App/Stats.hs

```diff
@@ -111,14 +115,33 @@ withStatsClient StatsSettings {..} f = do
   if amsEnabled
     then do
       tags <- (amsTags <>) <$> getEcsMetadataTags
-
+      gauges <- liftIO $ newIORef mempty
       Datadog.withDogStatsD amsSettings $ \client ->
         -- Add the tags to the thread context so they're present in all logs
         withThreadContext (map toPair tags)
-          $ f StatsClient { scClient = client, scTags = tags }
-    else f $ StatsClient { scClient = Datadog.Dummy, scTags = amsTags }
+          $ f StatsClient { scClient = client, scTags = tags, scGauges = gauges }
+    else do
+      gauges <- liftIO $ newIORef mempty
```

Comment:
If you're doing this in both branches, you could lift it up before the `if amsEnabled`.